sync2/handler2/handler.go

```diff
@@ -337,6 +340,10 @@ func (h *Handler) Initialise(ctx context.Context, roomID string, state []json.Ra
 
 func (h *Handler) SetTyping(ctx context.Context, roomID string, ephEvent json.RawMessage) {
 	next := typingHash(ephEvent)
+	// protect typingMap with a lock, so concurrent calls to SetTyping see the correct map
+	h.typingMu.Lock()
```

Comment:
Even though I couldn't get the test fail in CI, this was causing a race condition.

Reply:
All of the bits which touch `typingMap` or `typingDeviceHandler` should be protected.